src/components/ruler/Numbers.tsx

```diff
@@ -1,39 +1,35 @@
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import { useNumbersLineContext } from "../../context/numbersLineContext";
 import { LineRange } from "../../type/Line";
-
 interface IProps {
-  startIndex: number;
-  setStartIndex: (val: number) => void;
+  leftPosition: number;
 }
-
-const Numbers = ({ startIndex, setStartIndex }: IProps) => {
+const Numbers = ({ leftPosition }: IProps) => {
   const { kind } = useNumbersLineContext();
-  var endIndex = kind == LineRange.hundredCircular ? 101 : 21;
-
-  var labels = [];
-  if (kind == LineRange.hundredCircular)
-    labels = Array.from({ length: kind + 1 }, (_, index) => (index % 10 == 0 ? { value: index, isMainLine: true } : { value: index, isMainLine: false }));
-  else labels = Array.from({ length: kind }, (_, index) => ({ value: index, isMainLine: true }));
+  const [labels, setLabels] = useState<number[]>([]);
 
   useEffect(() => {
-    setStartIndex(0);
+    let array: number[] = [];
+    array = Array.from({ length: kind == LineRange.hundred ? kind + 1 : kind }, (_, index) => index);
+    setLabels(array);
```

Comment:
you can use it one line 
 let array: number[] =    array = Array.from({ length: kind == LineRange.hundred ? kind + 1 : kind }, (_, index) => index);